ping/tests/test_ping.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+import mock
+
+from datadog_checks.checks import AgentCheck
+from datadog_checks.ping import PingCheck
+from datadog_checks.errors import CheckException
+
+def mock_exec_ping():
+    return """FAKEPING 127.0.0.1 (127.0.0.1): 56 data bytes
+64 bytes from 127.0.0.1: icmp_seq=0 ttl=64 time=0.093 ms
+
+--- 127.0.0.1 ping statistics ---
+1 packets transmitted, 1 packets received, 0.0% packet loss
+round-trip min/avg/max/stddev = 0.093/0.093/0.093/0.000 ms"""
+
+@mock.patch("datadog_checks.ping.PingCheck._exec_ping", return_value=mock_exec_ping())
+def test_check(aggregator, instance):
+    c = PingCheck('ping', {}, {})
+
+    # empty instance
+    instance = {}
+    with pytest.raises(CheckException):
+        c.check(instance)
+
+    # only name
+    with pytest.raises(CheckException):
+        c.check({'name': 'Datadog'})
+
+
+    test_check
+    # good check
+    instance = {
+        'host': '127.0.0.1',
+        'name': "Localhost"
+    }
+    c.check(instance)
```

Comment:
An alternative approach to decorating the test function would be using `mock.patch` as a context manager within the function body. You have 2 alternatives:

1. mock the class **before** creating the instance, like you did by decorating the test function. This has the implication that every time you do `c = PingCheck('ping', {}, {})` **within** the `with` body, that specific `c` will result to be mocked.
2. mock the method **after** creating the instance, on the instance itself after you create it with `c = PingCheck('ping', {}, {})`

Option 2 might be more concise for this use case: 

```python
with mock.patch.object(c, "_exec_ping", return_value=mock_exec_ping()):
    c.check(instance)

aggregator.assert_service_check('network.ping.can_connect', AgentCheck.OK)
```
With the code above, the mock would be applied to the check **instance** and not the check class and the method will be mocked only within the `with` statement block, back to normal as soon as execution exits the block. So if you do `c.check({'name': 'Datadog'})` outside the block, `check` wouldn't be mocked and would behave as expected.